pkg/common/consumer/consumer_factory.go

```diff
@@ -94,6 +101,16 @@ func (c kafkaConsumerGroupFactoryImpl) startExistingConsumerGroup(
 	ctx, cancel := context.WithCancel(context.Background())
 
 	go func() {
+		// this is a blocking func
+		// do not proceed until the check is done
+		err := c.offsetsChecker.WaitForOffsetsInitialization(ctx, groupID, topics, logger, c.addrs, c.config)
+		if err != nil {
+			logger.Errorw("Error while checking if offsets are initialized", zap.Error(err))
```

Comment:
1. NIT error message starting with lowercase 
2. useful to add groupID and topics